type4py/learn_split.py

```diff
@@ -0,0 +1,353 @@
+import os
+
+from type4py.data_loaders import select_data, TripletDataset, load_training_data_per_model, \
+    load_training_data_per_model_sep
+from type4py.vectorize import AVAILABLE_TYPES_NUMBER, W2V_VEC_LENGTH
+from type4py.eval import eval_type_embed
+from type4py.utils import load_model_params
+from type4py import logger, MIN_DATA_POINTS, KNN_TREE_SIZE
+from torch.utils.data import DataLoader
+from typing import Tuple
+from collections import Counter
+from multiprocessing import cpu_count
+from os.path import join
+from time import time
+from annoy import AnnoyIndex
+from tqdm import tqdm
+import numpy as np
+import torch.nn as nn
+import torch
+import pickle
+import pkg_resources
+
+logger.name = __name__
+DEVICE = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+
+class ModelNotFit(Exception):
```

Comment:
Create a new module called `exceptions.py` and put all the exception classes there.